src/main/java/com/iota/iri/service/milestone/impl/MilestoneServiceImpl.java

```diff
@@ -151,6 +179,17 @@ public MilestoneValidity validateMilestone(TransactionViewModel transactionViewM
                                         transactionViewModel.getHash());
                                 newMilestoneViewModel.store(tangle);
 
+                                // if we find a NEW milestone that should have been processed before our latest solid
+                                // milestone -> reset the ledger state and check the milestones again
+                                //
+                                // NOTE: this can happen if a new subtangle becomes solid before a previous one while
+                                //       syncing
+                                if (milestoneIndex < snapshotProvider.getLatestSnapshot().getIndex() &&
+                                        milestoneIndex > snapshotProvider.getInitialSnapshot().getIndex()) {
+
+                                     resetCorruptedMilestone(newMilestoneViewModel.index());
```

Comment:
the use of `newMilestoneViewModel.index()` instead of `milestoneIndex` in this line made me look up the value and make sure they are the same -- it's better to use the same var for better readability.

```suggestion
                                     resetCorruptedMilestone(milestoneIndex);
```